miopen 6.3.0 fails to build due to missing `#include <cmath>`

On system with unmodified clang-19 with libstdc++-14 miopen-6.3.0 fails to build with:

```
In file included from /var/tmp/portage/sci-libs/miopen-6.3.0/work/MIOpen-rocm-6.3.0/driver/dm_reducecalculation.cpp:26:
/var/tmp/portage/sci-libs/miopen-6.3.0/work/MIOpen-rocm-6.3.0/driver/reducecalculation_driver.hpp:80:34: error: no matching function for call to 'isnan'
   80 |             if(nanPropagation && isnan(val))
      |                                  ^~~~~
```

Note, that on some other systems it compiles for various reasons. For example, on libc++ systems it works fine, as `math.h` is indirectly included in `<algorithm>`
```
#include <algorithm>
// #include <math.h>

auto x = isnan(1);
```
Conformance viewer: https://godbolt.org/z/qvG61vE8W

Could you apply this fix? Thanks!

```diff
--- a/driver/reducecalculation_driver.hpp
+++ b/driver/reducecalculation_driver.hpp
@@ -34,6 +34,7 @@
 #include <algorithm>
 #include <cfloat>
 #include <cstdlib>
+#include <cmath>
 #include <memory>
 #include <miopen/miopen.h>
 #include <miopen/tensor.hpp>
@@ -77,7 +78,7 @@ int32_t mloReduceCalculationForwardRunHost(miopenTensorDescriptor_t inputDesc,
         for(size_t i = 0; i < reduce_size; ++i)
         {
             Tcheck val = static_cast<Tcheck>(input[input_idx]);
-            if(nanPropagation && isnan(val))
+            if(nanPropagation && std::isnan(val))
             {
                 val = 0.0f;
             }

```

--- a/driver/reducecalculation_driver.hpp
+++ b/driver/reducecalculation_driver.hpp
@@ -33,6 +33,7 @@
 #include "random.hpp"
 #include <algorithm>
 #include <cfloat>
+#include <cmath>
 #include <cstdlib>
 #include <memory>
 #include <miopen/miopen.h>
@@ -77,7 +78,7 @@ int32_t mloReduceCalculationForwardRunHost(miopenTensorDescriptor_t inputDesc,
         for(size_t i = 0; i < reduce_size; ++i)
         {
             Tcheck val = static_cast<Tcheck>(input[input_idx]);
-            if(nanPropagation && isnan(val))
+            if(nanPropagation && std::isnan(val))
             {
                 val = 0.0f;
             }

